### PR TITLE
PB-103 : add parsing of getFeatureInfo capabilities for external layers

### PR DESCRIPTION
**Pull request by pakb**
[Test link](https://sys-map.dev.bgdi.ch/preview/feat_pb-103_getfeatureinfo_cap_parsing/index.html)

### CODE REVIEW

File: src/api/layers/ExternalLayer.class.js [hunk 1]
```diff
@@ -117,6 +140,11 @@ export default class ExternalLayer extends AbstractLayer {
         this.abstract = abstract
         this.extent = extent
         this.legends = legends
+        this.availableProjections = [...availableProjections]
+        if (this.availableProjections.length === 0) {
+            this.availableProjections.push(WGS84)
+        }
```
Comment (ltshb) [inline src/api/layers/ExternalLayer.class.js:146]: Might be wise to add a log warning here that we have a fallback to WGS84

File: src/api/layers/WMSCapabilitiesParser.class.js [hunk 2]
```diff
@@ -42,6 +42,43 @@ export default class WMSCapabilitiesParser {
...
+                baseUrl = httpElement.Post.OnlineResource
+            } else {
+                log.error(
+                    "Couldn't parse GetFeatureInfo data",
+                    this.Capability.Request.GetFeatureInfo
+                )
+                if (ignoreError) {
+                    return null
+                }
+                throw new CapabilitiesError('Invalid GetFeatureInfo data')
```
Comment (ltshb) [inline src/api/layers/WMSCapabilitiesParser.class.js:71]: Here I would add a translation key after the message

```javascript
throw new CapabilitiesError('Invalid GetFeatureInfo data', 'invalid_get_feature_info')
```

We could use it then later for displaying an error

File: src/api/layers/WMSCapabilitiesParser.class.js [hunk 3]
```diff
@@ -42,6 +42,43 @@ export default class WMSCapabilitiesParser {
...
+            }
+            return {
+                baseUrl,
+                method,
+                formats: [...this.Capability.Request.GetFeatureInfo.Format],
```
Comment (ltshb) [inline src/api/layers/WMSCapabilitiesParser.class.js:76]: What if `Format` is undefined ? Maybe `...this.Capability.Request.GetFeatureInfo.Format ?? {}`

**Review by ltshb — APPROVED** [review]
Good just minor comments. 

You should also add some unitest for the new get feature info capabitlies

File: src/api/layers/WMSCapabilitiesParser.class.js (hunk 3 above)
Comment (pakb) [reply src/api/layers/WMSCapabilitiesParser.class.js:76]: As I'm using this array later and wouldn't want "default" value such as an empty Object, I'll check that Format exists before pushing it into the formats array.
This way, the formats array will stay empty if no format is defined by the service

**Pull request merged**